pkg/providers/cloudstack/cloudstack.go

```diff
@@ -280,65 +271,21 @@ func (p *cloudstackProvider) MachineResourceType() string {
 	return eksaCloudStackMachineResourceType
 }
 
-func (p *cloudstackProvider) setupSSHAuthKeysForCreate() error {
-	var useKeyGeneratedForControlplane, useKeyGeneratedForWorker bool
-	var err error
-	controlPlaneUser := p.machineConfigs[p.clusterConfig.Spec.ControlPlaneConfiguration.MachineGroupRef.Name].Spec.Users[0]
-	if len(controlPlaneUser.SshAuthorizedKeys[0]) > 0 {
-		controlPlaneUser.SshAuthorizedKeys[0], err = common.StripSshAuthorizedKeyComment(controlPlaneUser.SshAuthorizedKeys[0])
-		if err != nil {
-			return err
-		}
-	} else {
-		p.log.Info("Provided control plane sshAuthorizedKey is not set or is empty, auto-generating new key pair...")
-		generatedKey, err := common.GenerateSSHAuthKey(p.writer)
-		if err != nil {
-			return err
-		}
-		controlPlaneUser.SshAuthorizedKeys[0] = generatedKey
-		useKeyGeneratedForControlplane = true
-	}
-	var workerUser v1alpha1.UserConfiguration
-	for _, workerNodeGroupConfiguration := range p.clusterConfig.Spec.WorkerNodeGroupConfigurations {
-		workerUser = p.machineConfigs[workerNodeGroupConfiguration.MachineGroupRef.Name].Spec.Users[0]
-		if len(workerUser.SshAuthorizedKeys[0]) > 0 {
-			workerUser.SshAuthorizedKeys[0], err = common.StripSshAuthorizedKeyComment(workerUser.SshAuthorizedKeys[0])
-			if err != nil {
-				return err
-			}
-		} else {
-			if useKeyGeneratedForControlplane { // use the same key
-				workerUser.SshAuthorizedKeys[0] = controlPlaneUser.SshAuthorizedKeys[0]
-			} else {
-				p.log.Info("Provided worker sshAuthorizedKey is not set or is empty, auto-generating new key pair...")
-				generatedKey, err := common.GenerateSSHAuthKey(p.writer)
-				if err != nil {
-					return err
-				}
-				workerUser.SshAuthorizedKeys[0] = generatedKey
-				useKeyGeneratedForWorker = true
-			}
-		}
-	}
-	if p.clusterConfig.Spec.ExternalEtcdConfiguration != nil {
-		etcdUser := p.machineConfigs[p.clusterConfig.Spec.ExternalEtcdConfiguration.MachineGroupRef.Name].Spec.Users[0]
-		if len(etcdUser.SshAuthorizedKeys[0]) > 0 {
-			etcdUser.SshAuthorizedKeys[0], err = common.StripSshAuthorizedKeyComment(etcdUser.SshAuthorizedKeys[0])
-			if err != nil {
-				return err
-			}
-		} else {
-			if useKeyGeneratedForControlplane { // use the same key as for controlplane
-				etcdUser.SshAuthorizedKeys[0] = controlPlaneUser.SshAuthorizedKeys[0]
-			} else if useKeyGeneratedForWorker {
-				etcdUser.SshAuthorizedKeys[0] = workerUser.SshAuthorizedKeys[0] // if cp key was provided by user, check if worker key was generated by cli and use that
-			} else {
-				p.log.Info("Provided etcd sshAuthorizedKey is not set or is empty, auto-generating new key pair...")
-				generatedKey, err := common.GenerateSSHAuthKey(p.writer)
+func (p *cloudstackProvider) generateSSHKeysIfNotSet(machineConfigs map[string]*v1alpha1.CloudStackMachineConfig) error {
```

Comment:
Do we still need this if users are required to provide ssh keys?

Reply:
To clarify, the ssh keys are required for the template builder. There is no validation blocking users from not entering SSH keys. My goal here was just to remove the stripping of comments from the ssh keys, as that is new done in the template builder and preserving the rest of current CLI behavior.

Reply:
So are we going to remove `generateSSHKeysIfNotSet` once we add validation blocking users from not entering SSH keys? If so, could you sync with mitali and also leave a comment there?

Reply:
I don't think we're removing the function of generating ssh keys in the CLI anytime soon. We've only decided to block it in the webhook if an SSH key isn't specified. The team hasn't made a decision of how we'll achieve parity between the API and CLI as far as I know. I will double check with @mitalipaygude though